### PR TITLE
Monitoring and healthchecks

### PR DESCRIPTION
**Pull request by benjaminbillet**
Hello,

Here is a PR for:
- replacing metrics-core by micrometer, and adding a prometheus endpoint
- adding healthchecks endpoints
- tests with kafka unit
- bugfix: handle properly "enable topic delete" flag
- configurable timeout for kafka communication

Regards

### CODE REVIEW

File: src/main/java/nb/kafka/operator/HealthServer.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    PathHandler pathHandler = new PathHandler();
+    pathHandler.addPrefixPath("/healthy", exchange -> exchange.setStatusCode(StatusCodes.OK));
+    pathHandler.addPrefixPath("/ready", exchange -> {
+      if (isKafkaAvailable(config))
+        exchange.setStatusCode(StatusCodes.OK);
+      else
+        exchange.setStatusCode(StatusCodes.INTERNAL_SERVER_ERROR);
+    });
+    logger.info("Starting health checks server on port: " + DEFAULT_PORT);
+    Undertow server = Undertow.builder().addHttpListener(DEFAULT_PORT, DEFAULT_HOST)
```
Comment (nbogojevic) [inline src/main/java/nb/kafka/operator/HealthServer.java]: Do we need undertow here? Can we use something simpler like the one already coming from Prometheus exporter (HTTPServer)?

Comment (benjaminbillet) [reply src/main/java/nb/kafka/operator/HealthServer.java]: The problem with io.prometheus.client.exporter.HTTPServer is that it is not a general purpose web server, we can only give it the prometheus registry and there is no method to create custom endpoints on it.

Comment (nbogojevic) [reply src/main/java/nb/kafka/operator/HealthServer.java]: As discussed offline, will be using com.sun.net.httpserver.HttpServer

Comment (roufa85) [reply src/main/java/nb/kafka/operator/HealthServer.java]: Hi @nbogojevic , I have used undertow as a lightweight http server. I got your point to reduce the number of dependencies. Implement health server using prometheus will bring less code readability and that would mix up the different usage of endpoints in the same code base.
In our last [commit](https://github.com/benjaminbillet/kafka-operator/commit/add99727093cc17fcaad71e3fafb1fd3b13b2209#diff-4421998a9e51f7f27d441bc5c712503fR14), we have implemented the HealthCheckServer using  `com.sun.net.httpserver.HttpServer`.
We have removed undertow dependency.

**Pull request merged**